swim/join_sender.go

```diff
@@ -334,6 +398,68 @@ func (j *joinSender) JoinCluster() ([]string, error) {
 	return nodesJoined, nil
 }
 
```

Comment:
yea, I thought about an abstraction like that and decided not to go that far initially. the more I think about it, the more I like it. I'll give it a go.